### ISSUE TITLE
Is this a bug in the loss function

### ISSUE THREAD

**Issue by NicolasHoferer** [AidenDurrant/SCL-Pytorch#1]
Hey Aiden,

first of all thank you for your great implementation.

I'm confused about the computation of the cost function.
As far as I understood you didn't count the cost contribution of the image j which comes from the same anchor as image i, i.e. i and j are both augmented versions of the same image.
I think you don't count it in both sums, the outer and the inner sum over all pairs.

Regarding the inner sum the authors wrote:
> for any anchor, all positives in a minibatch (i.e. the augmentation-based one as well as any of the remaining 2(N-1) entries that are from the same class) contribute to the numerator.

And regarding the outer sum, from the formula I understand they add the contribution for L_i from the augmentation based j, since they share the same class.

One easy example is just taking two random inputs for z_i and z_j of batch size 4 and targets = [0,1,2,3], then your loss would output 0.0.

I hope you understand my explanation.

Thank you for your thoughts on that

Nic

**Comment by AidenDurrant** [comment]
Hi Nic,

Thanks for getting in contact with me, I will have a look now at my code
tonight and get back to you with an answer. Off the top of my head, I'm
pretty sure I included the computation of *I* and *J* but did not include
the computation of *J* and *I* as they are the same. I will, of course,
check this and update the code with any changes.

I really appreciate your time to email, thanks again.

Best Regards,

Aiden
